pep_sphinx_extensions/pep_processor/transforms/pep_references.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+
+from docutils import nodes
+from docutils import transforms
+
+
+class PEPReferenceRoleTitleText(transforms.Transform):
+    """Add title text of document titles to reference role references."""
+
+    default_priority = 730
+
+    def apply(self) -> None:
+        if not Path(self.document["source"]).match("pep-*"):
+            return  # not a PEP file, exit early
+        for node in self.document.findall(nodes.reference):
+            if "_title_tuple" not in node:
+                continue
+
+            # get pep number and section target (fragment)
+            pep_num, fragment = node.attributes.pop("_title_tuple")
+            filename = f"pep-{pep_num:0>4}"
+
+            # Cache target_ids
+            env = self.document.settings.env
+            try:
+                target_ids = env.document_ids[filename]
+            except KeyError:
+                env.document_ids[filename] = target_ids = env.get_doctree(filename).ids
+
+            # Create title text string. We hijack the 'reftitle' attribute so
+            # that we don't have to change things in the HTML translator
+            node["reftitle"] = env.titles[filename].astext().replace(" – ", ", ")
```

Comment:
```suggestion
            node["reftitle"] = env.titles[filename].astext()
```

Is there a reason ` - ` isn't just retained here, to be consistent with actual displayed PEP titles, and a comma is substituted instead? I'm not sure how this is motivated by the semantics here; if we do need to pick a different character, a colon would be more appropriate.

Reply:
Yes, but context is important here. The footnotes contained commas because they were formatted as _citations_, whereas this is used as _title-text_ of a link, which is intended to display the _title_ of the linked document—which uses a dash. So I think we should just follow precedent and be consistent with the actual title, leaving it as-is.

I don't really see the issue with appending the section, since you were thoughtful to clearly separate it with the unique and unambiguous section symbol (`§`) and considering the standard format is `title § section`, it would be odd to not use the actual title.